src/Functions/FunctionsHashing.h

```diff
@@ -1025,17 +1025,58 @@ class FunctionAnyHash : public IFunction
 
                 if constexpr (Impl::use_int_hash_for_pods)
                 {
-                    if constexpr (std::is_same_v<ToType, UInt64>)
-                        h = IntHash64Impl::apply(bit_cast<UInt64>(vec_from[i]));
+                    if constexpr (std::endian::native == std::endian::little)
```

Comment:
I think the code will be ca. 12.8% more aesthetic if it is structured like that:

```cpp
ToType h;

if constexpr (std::is_same_v<ToType, UInt64>)
{
    UInt64 v = bit_cast<Uint64>(vec_from[i]);
    if (std::endian::native == std::endian::big)
        v = __builtin_bswap64(v);
    h = IntHash64Impl::apply(v);
}
else
{
    UInt32 v = bit_cast<Uint32>(vec_from[i]);
    if (std::endian::native == std::endian::big)
        v = __builtin_bswap32(v);
    h = IntHash32Impl::apply(v);
}
```

Reply:
Modified title and changelog.
Also refactored code according to previous comments.